src/Corporation/helpers.ts

```diff
@@ -12,6 +13,15 @@ export function calculateUpgradeCost(corporation: Corporation, upgrade: CorpUpgr
   return cost;
 }
 
+export function calculateOfficeSizeUpgradeCost(currentSize: number, sizeIncrease: number): number {
```

Comment:
I think it's worth the effort to be more specific here.

Reply:
I went further and used `PositiveInteger`, but according to my understanding `sizeIncrease: PositiveInteger` doesn't actually validate the value of `sizeIncrease` like `helpers.positiveInteger` does, so I'm leaving this check in place.

Reply:
The correct thing to do with these types is to (runtime) check them *as* they are TS-asserted as `PositiveInteger` or `Integer`. The type assertion functions in `types.ts` do both - they take a generic `number` and assert that it has the proper form if the function returns true. You are expected to return/throw/otherwise abort flow-control if false, and then TS can know that the asserted-true path is also of the more restrictive type.

As a result, anything that is declared `PositiveInteger` *should have already be runtime checked as a positive integer.* Anything else is a bug.

Reply:
The loop one _should_ be ok because the while condition is the strict inequality `upgradeSizeMax > props.office.size` so `upgradeSizeMax - props.office.size` should always be positive.

Reply:
But props.office.size could be a non-integer. (It shouldn't be, but the type system doesn't have anything to say about it.) I'm this case that's fine because the formula actually functions fine with non-integer sizeIncrease, even if that were to happen.